### PR TITLE
Update makefile with a release-manifests target

### PR DESCRIPTION
**Pull request by thunderboltsid**
The target generates infrastructure-components.yaml and cluster templates and puts them in the `RELEASE_DIR` that defaults to `out`.

### CODE REVIEW

**Review by yannickstruyf3 — APPROVED** [review]

**Review by deepakm-ntnx — COMMENTED** [review]
could u pls update github workflows file too to call this?

**Comment by thunderboltsid** [comment]
> could u pls update github workflows file too to call this?

We can do that in another PR; not sure if our release process expects the github workflow.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -95,6 +95,6 @@ jobs:
         generate_release_notes: true
         append_body: true
         files: |
-          infrastructure-components.yaml
+          out/infrastructure-components.yaml
           metadata.yaml
           templates/cluster-template*.yaml
```
Comment (deepakm-ntnx) [inline .github/workflows/release.yaml:100]: out or templates folder should we take this file from? can we consolidate?

Comment (thunderboltsid) [reply .github/workflows/release.yaml:100]: templates is the versioned directory; out it a gitignored directory.

**Review by deepakm-ntnx — DISMISSED** [review]

**Review by yannickstruyf3 — DISMISSED** [review]

**Review by deepakm-ntnx — APPROVED** [review]

**Review by yannickstruyf3 — APPROVED** [review]

**Review by tuxtof — DISMISSED** [review]

**Review by yannickstruyf3 — APPROVED** [review]

**Review by yannickstruyf3 — APPROVED** [review]

**Review by tuxtof — APPROVED** [review]

**Pull request merged**